### PR TITLE
Metadata manager bugfixes

### LINKED ISSUE #6629: When I upload a model with one to many I cannot make an attribute visible through metadata editing
### How to Reproduce
Upload 
[meta_data.zip](https://github.com/molgenis/molgenis/files/1296113/meta_data.zip)
Go to the metadata manager
Select eu_bbmri_eric_collections
Select standards
Check visible
Save

### Expected behavior
Saved!

### Observed behavior
Invalid [xref] value [] for attribute [Mapped by] of entity [sub_collections] with type [sys_md_Attribute]. Offended expression: $('mappedBy').isNull().and($('type').eq('onetomany').not()).or($('mappedBy').isNull().not().and($('type').eq('onetomany'))).value()

### LINKED ISSUE #6736: Metadata editor does not show select box for reference entity when attr type is mref
### How to Reproduce
- Login to molgenis ( admin)
- Navigate to the the metadata manager plugin 
- Use the 'plus' button to create a new entity
- Create a new attribute. and set the 'Type' field to 'CATEGORICALMREF'

### Expected behavior
The 'Reference entity' entity select becomes visible.

### Observed behavior
The 'Reference entity' select does **not** becomes visible.

### PR DESCRIPTION
**Pull request by sidohaakma**
* Fix #6629 When I upload a model with one to many I cannot make an attribute visible through metadata editing
* Fix #6736: Metadata editor does not show select box for reference entity when attr type is mref
* Workaround #6739: Metadata manager cannot store order by column
  (Works by showing the order by if present)

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits

### CODE REVIEW

**Comment by sidohaakma** [comment]
@fdlk, could you look at the Vue Component test? I think this implementation will work. The selectedAttribute in the Vue-component can also be addressed to read the properties of refEntityType and mappedByAttribute. If you address the properties via the selectedAttribute and use the "expect"-method in the manner it is used now, the test works. I implemented the selectedAttribute in all the "expect"-method calls.

**Comment by sidohaakma** [comment]
Before someone looks at this I will first fix the unit tests!

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue [hunk 1]
```diff
@@ -255,6 +255,7 @@
         }).catch(this.$swal.noop)
       },
       onAttributeSelect (selectedAttribute) {
+        console.log(selectedAttribute)
```
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: Please remove console log

File: molgenis-metadata-manager/src/main/frontend/src/store/getters.js [hunk 2]
```diff
@@ -64,6 +64,13 @@ export default {
     return rootAttributes.map(addChildren)
   },
   /**
+   *
+   */
+  getMappedByAttributes: (state, getters) => {
```
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/store/getters.js]: You do not need the getters here.
can also write as `getMappedByAttributes: state => state.initialEditorEntityType.referringAttributes`

**Comment by Mark-de-Haan** [comment]
## Error

`Could not read JSON: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 772; nested exception is com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 772`

## Reproduce in the Metadata manager
- Create new Entity OneToManyParent
add attribute: id
save -> ok

- Create new Entity OneToManyXref
add attributes: id, xref -> OneToManyParent
save -> ok

- Edit Entity OneToManyParent
add attribute: oneToMany -> OneToManyXref - xref
save -> error

File: molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/service/MetadataManagerServiceImpl.java [hunk 3]
```diff
@@ -45,15 +54,20 @@ public MetadataManagerServiceImpl(MetaDataService metadataService, PackageMapper
...
+
 		if (entityType == null)
 		{
 			throw new UnknownEntityException("Unknown EntityType [" + entityTypeId + "]");
 		}
-
-		return createEntityTypeResponse(entityType);
+		Repository<Attribute> attributeRepository = metadataService.getRepository(ATTRIBUTE_META_DATA,
+				Attribute.class);
+		//TODO: en nu zonder repository
```
Comment (fdlk) [inline molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/service/MetadataManagerServiceImpl.java]: TODO!

File: molgenis-metadata-manager/src/test/java/org/molgenis/metadata/manager/service/MetadataManagerServiceTest.java [hunk 4]
```diff
@@ -166,7 +185,7 @@ private EditorAttributeResponse getEditorAttributeResponse()
 		@Bean
 		public MetaDataService metaDataService()
 		{
-			return mock(MetaDataService.class);
+			return mock(MetaDataService.class, RETURNS_DEEP_STUBS);
```
Comment (fdlk) [inline molgenis-metadata-manager/src/test/java/org/molgenis/metadata/manager/service/MetadataManagerServiceTest.java]: What's a deep stub and why do I need one?

**Comment by marikaris** [comment]
I try to reproduce #6629 now but most of my attributes are gone in the metadatamanager (but not in the dataexplorer)
screen shot 2017-09-29 at 14 17 50
screen shot 2017-09-29 at 14 17 34
screen shot 2017-09-29 at 14 17 18

File: molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue [hunk 5]
```diff
@@ -39,7 +39,7 @@
     },
     computed: {
       isFolder: function () {
-        return this.attribute.type === 'COMPOUND'
+        return this.attribute.type === 'compound'
```
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue]: huh? When the attribute types get retrieved from the server, they are toUpperCased() before being put in the state....

**Pull request merged**